pkg/controller/argocd/secret.go

```diff
@@ -379,6 +385,38 @@ func (r *ReconcileArgoCD) reconcileGrafanaSecret(cr *argoprojv1a1.ArgoCD) error
 	return r.client.Create(context.TODO(), secret)
 }
 
+// reconcileClusterPermissionsSecret ensures ArgoCD instance is namespace-scoped
+func (r *ReconcileArgoCD) reconcileClusterPermissionsSecret(cr *argoprojv1a1.ArgoCD) error {
+	secret := argoutil.NewSecretWithSuffix(cr.ObjectMeta, "namespaces")
+	secret.Labels[common.ArgoCDClusterSecretLabel] = "cluster"
+	dataBytes, _ := json.Marshal(map[string]interface{}{
+		"tlsClientConfig": map[string]interface{}{
+			"insecure": false,
+		},
+	})
+
+	secret.Data = map[string][]byte{
+		"config":     dataBytes,
+		"name":       []byte("in-cluster"),
+		"server":     []byte(common.ArgoCDDefaultServer),
+		"namespaces": []byte(cr.Namespace),
+	}
+
+	applyReconcilerHook(cr, secret, "")
+
+	existingSecret := &corev1.Secret{}
+	if argoutil.IsObjectFound(r.client, secret.Namespace, secret.Name, existingSecret) {
+		if reflect.DeepEqual(secret.Data["namespaces"], existingSecret.Data["namespaces"]) {
+			// do nothing
+			return nil
+		}
+		existingSecret.Data["namespaces"] = secret.Data["namespaces"]
+		return r.client.Update(context.TODO(), existingSecret)
+	}
```

Comment:
Actually, this would prevent users from extending the scope of the Argo CD instance, right? So, it will prevent users from manually adding namespaces to be managed, but instead always reconcile the namespace back to the name of the namespace the Argo CD instance is running in. I think we _should_ let users override this, if they really would like to.

Reply:
Yeah, this is what i was confused with. But if we are providing an alternative that would create a new Secret, should we be allowing the user to update this manually?

Reply:
done !